tests/Serval.SpecFlowTests/StepDefinitions/MachineApiStepDefinitions.cs

```diff
@@ -0,0 +1,337 @@
+        string targetLanguage,
+        bool pretranslate
+    )
+    {
+        if (!Enum.TryParse(fileFormatString, ignoreCase: true, result: out FileFormat fileFormat))
+            throw new ArgumentException(
+                "Corpus format type needs to be one of: " + string.Join(", ", EnumToStringList<FileFormat>())
+            );
+        var filesToAdd = filesToAddString.Split(", ");
+        var engineId = await GetEngineFromUser(user);
+
+        var corpusId = await PostCorpusToEngine(
+            engineId,
+            fileFormat: fileFormat,
+            filesToAdd,
+            sourceLanguage,
+            targetLanguage,
+            pretranslate
+        );
+    }
+
+    [When(@"(.*)'s engine is built")]
+    public async Task WhenEngineIsBuild(string user)
+    {
+        var engineId = await GetEngineFromUser(user);
+        var newJob = await translationEnginesClient.StartBuildAsync(engineId, translationBuildConfig);
+        int cRevision = newJob.Revision;
+        while (true)
+        {
+            var result = await translationEnginesClient.GetBuildAsync(engineId, newJob.Id);
+            if (!(result.State == JobState.Active || result.State == JobState.Pending))
+            {
+                // build completed
+                break;
+            }
+            Thread.Sleep(500);
+        }
+    }
+
+    [When(@"a translation for (.*) is added with ""(.*)"" for ""(.*)""")]
+    public async Task WhenTranslationAdded(string user, string targetSegment, string sourceSegment)
+    {
+        var engineId = await GetEngineFromUser(user);
+        await translationEnginesClient.TrainSegmentAsync(
+            engineId,
+            new SegmentPair
+            {
+                SourceSegment = sourceSegment,
+                TargetSegment = targetSegment,
+                SentenceStart = true
+            }
+        );
+    }
+
+    [When(@"the translation for (.*) for ""(.*)"" is ""(.*)""")]
+    public async Task WhenTheTranslationIs(string user, string sourceSegment, string targetSegment)
+    {
+        await ThenTheTranslationShouldBe(user, sourceSegment, targetSegment);
+    }
+
+    [Then(@"the translation for (.*) for ""(.*)"" should be ""(.*)""")]
+    public async Task ThenTheTranslationShouldBe(string user, string sourceSegment, string targetSegment)
+    {
+        var engineId = await GetEngineFromUser(user);
+        var translation = await translationEnginesClient.TranslateAsync(engineId, sourceSegment);
+        Assert.AreEqual(targetSegment, translation.Translation);
+    }
+
+    [Then(@"the pretranslation for (.*) for (.*) starts with ""(.*)""")]
+    public async Task ThenThePretranslationShouldBe(string user, string fileId, string targetSegment)
+    {
+        var engineId = await GetEngineFromUser(user);
+        if (translationBuildConfig.Pretranslate == null | translationBuildConfig.Pretranslate!.Count == 0)
+        {
+            throw new Exception("Need to have something to pretranslate!");
```

Comment:
`InvalidOperationException` would be a more appropriate exception to use here.

Reply:
Done.